BungeeCord-Patches/0067-Always-slice-until-entity-rewrite-is-being-used-in-M.patch

```diff
@@ -0,0 +1,57 @@
+From fa7dcf4b0fa0b5aaf20c9071b547699ae43ab530 Mon Sep 17 00:00:00 2001
+From: BoomEaro <21033866+BoomEaro@users.noreply.github.com>
+Date: Mon, 3 Apr 2023 13:03:53 +0300
+Subject: [PATCH] Always slice until entity rewrite is being used in
+ MinecraftDecoder
+
+
+diff --git a/protocol/src/main/java/net/md_5/bungee/protocol/MinecraftDecoder.java b/protocol/src/main/java/net/md_5/bungee/protocol/MinecraftDecoder.java
+index ac83e325..b6950ff1 100644
+--- a/protocol/src/main/java/net/md_5/bungee/protocol/MinecraftDecoder.java
++++ b/protocol/src/main/java/net/md_5/bungee/protocol/MinecraftDecoder.java
+@@ -20,6 +20,8 @@ public class MinecraftDecoder extends MessageToMessageDecoder<ByteBuf>
+     private int protocolVersion;
+     @Setter
+     private boolean supportsForge = false;
++    @Setter
++    private boolean slice = true; // Waterfall
```

Comment:
> Shouldn't we default to the safe side, copy instead of slice?

As far as I remember, the only situation where the proxy actually changes the buffer is when an entity map instance appears in the UserConnection.
Before the proxy creates the UserConnection, the proxy decodes the Handshake, Login, and a couple of other packets. (when the state is not yet FINISHING in InitialHandler)
It doesn't seem to make sense at this stage specifically to copy the buffer for these packets.

Reply:
btw why you don't set this directly at initialization of minecraftdecoder to the config value of rewriting disabled?